### PR TITLE
Overhauled documentation, moved ensemble models to core, various fixes

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- [x] Moved abstract model to core
- [x] Moved ensemble models to core
- [x] Added view source code option in docs for classes: [Example](http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-893/7/_modules/autogluon/core/models/abstract/abstract_model.html#AbstractModel)
- [x] Added autogluon.core.models documentation: [Example](http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-893/7/api/autogluon.tabular.models.html#module-autogluon.core.models)
- [x] Added autogluon.tabular.models documentation: [Example](http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-893/7/api/autogluon.tabular.models.html#module-autogluon.tabular.models)
- [x] Added autogluon.features documentation: [Example](http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-893/7/api/autogluon.features.html)
- [x] Added positive_class option to TabularPredictor
- [x] Fixed CatBoost for F1, Recall, Precision metric variants
- [x] Moved label_cleaner and cleaner to core
- [x] Various minor cleanup

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-893-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-893/1/index.html

**Comment by szha** [comment]
Job PR-893-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-893/2/index.html

**Comment by szha** [comment]
Job PR-893-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-893/3/index.html

**Comment by szha** [comment]
Job PR-893-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-893/4/index.html

**Comment by szha** [comment]
Job PR-893-7 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-893/7/index.html

File: tabular/src/autogluon/tabular/predictor/predictor.py [hunk 1]
```diff
@@ -82,6 +81,13 @@ class TabularPredictor(TabularPredictorV1):
         learner_kwargs : dict, default = None
             Kwargs to send to the learner. Options include:
 
+            positive_class : str or int, default = None
+                Used to determine the positive class in binary classification.
+                This is used for certain metrics such as 'f1' which produce different scores depending on which class is considered the positive class.
+                If not set, will be inferred as the second element of the existing unique classes after sorting them.
+                    If classes are [0, 1], then 1 will be selected as the positive class.
+                    If classes are ['def', 'abc'], then 'def' will be selected as the positive class.
+                    If classes are [True, False], then True will be selected as the positive class.
```
[the remark this answers is not in the document]
Comment (Innixma) [reply tabular/src/autogluon/tabular/predictor/predictor.py:90]: These are examples of the logic, it is based on sorting. Yes / Y would be positive class due to sorting.

File: tabular/src/autogluon/tabular/trainer/abstract_trainer.py [hunk 2]
```diff
@@ -1755,62 +1753,8 @@ def save_info(self, include_model_info=False):
...
-                    models_expanded += valid_models
-
-                hyperparameters_valid[key][subkey] = models_expanded
-        if 'default' not in hyperparameters_valid.keys():
-            level_keys = [key for key in hyperparameters_valid.keys() if isinstance(key, int)]
-            max_level_key = max(level_keys)
-            hyperparameters_valid['default'] = copy.deepcopy(hyperparameters_valid[max_level_key])
-        return hyperparameters_valid
+    def _process_hyperparameters(self, hyperparameters: dict) -> dict:
+        return process_hyperparameters(hyperparameters=hyperparameters)
```
[the remark this answers is not in the document]
Comment (Innixma) [reply tabular/src/autogluon/tabular/trainer/abstract_trainer.py:1757]: possibly, it theoretically gives flexibility to extensions of abstract_trainer to override the functionality. I don't think it is high priority to remove for v0.1.

File: core/src/autogluon/core/data/label_cleaner.py [hunk 3]
```diff
@@ -144,37 +144,31 @@ def _generate_categorical_mapping(y: Series) -> dict:
...
         y = self._convert_to_valid_series(y)
         self.num_classes = 2
         self.unique_values = list(y.unique())
         if len(self.unique_values) != 2:
             raise AssertionError('y does not contain exactly 2 unique values:', self.unique_values)
+        try:
+            self.unique_values.sort()
+        except TypeError:
+            # Contains both str and int
+            self.unique_values = list(y.unique())
```
[the remark this answers is not in the document]
Comment (Innixma) [reply core/src/autogluon/core/data/label_cleaner.py:158]: All of these situations are currently handled correctly in general due to the sorting N,Y n,y f,t false,true 0,1 etc. We can look to more hardcoded rules later on if necessary though.

**Review by gradientsky — APPROVED** [review]

**Pull request merged**